pkgs/development/libraries/aspell/dictionaries.nix

```diff
@@ -1,137 +1,217 @@
-{stdenv, fetchurl, aspell, which}:
+{lib, stdenv, fetchurl, aspell, which}:
+
+with lib;
+
+/* HOWTO:
+
+   * Add some of these to your profile or systemPackages.
+
+     ~~~~
+     environment.systemPackages = [
+       aspellDicts.en
+       aspellDicts.en-computers
+       aspellDicts.en-science
+     ];
+     ~~~~
+
+   * Rebuild and switch to the new profile.
+   * Add something like
+
+     ~~~~
+     master en_US
+     extra-dicts en-computers.rws
+     add-extra-dicts en_US-science.rws
+     ~~~~
+
+     to `/etc/aspell.conf` or `~/.aspell.conf`.
+   * Check that `aspell -a` starts without errors.
+   * (optional) Check your config with `aspell dump config | grep -vE '^(#|$)'`.
+   * Enjoy.
+
+*/
 
 let
 
-  /* Function to compile an Aspell dictionary.  Fortunately, they all
+  /* Function to compile an Aspell language dictionary.  Fortunately, they all
      build in the exact same way. */
-  buildDict =
-    {shortName, fullName, src, postInstall ? ""}:
+  buildLang =
+    {shortName, fullName, ...}@args:
 
-    stdenv.mkDerivation {
+    stdenv.mkDerivation ({
       name = "aspell-dict-${shortName}";
 
-      inherit src;
-
       buildInputs = [aspell which];
 
       dontAddPrefix = true;
 
       preBuild = "makeFlagsArray=(dictdir=$out/lib/aspell datadir=$out/lib/aspell)";
 
-      inherit postInstall;
-
       meta = {
         description = "Aspell dictionary for ${fullName}";
         platforms = stdenv.lib.platforms.all;
-      };
-    };
+      } // (args.meta or {});
+    } // removeAttrs args [ "meta" ]);
 
-in {
-
-  ca = buildDict {
+  /* Function to compile txt dict files into Aspell dictionaries. */
+  buildDict =
```

Comment:
How about creating instead a `src` that can be consumed by the old `buildDict? Any reason for not doing that?